org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java

```diff
@@ -387,8 +463,83 @@ public boolean accept(File dir, String name) {
 		return shouldSynchronize;
 	}
 
+	public static boolean upgradeGradleVersion(String projectDir, IProgressMonitor monitor) {
+		String newDistributionUrl = String.format("https://services.gradle.org/distributions/gradle-%s-bin.zip", GradleCompatibilityChecker.CURRENT_GRADLE);
+		Path projectFolder = Paths.get(projectDir);
```

Comment:
projectDir is passed as a uri string, does`Paths.get(uriString)` support uri string?

Reply:
Currently it's a path string actually. I'll correct the naming of this variable.

Reply:
fixed in https://github.com/eclipse/eclipse.jdt.ls/pull/1965/commits/87298b495b093411006826cc69f1775704fea69d